[BUG] Long flow names cause a panic during plan/apply

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### DaVinci Terraform provider Version
<!--- Check the version you have configured in your .tf files. If you are not running the latest version of the provider, please upgrade because your issue may have already been fixed. -->
0.1.13

### Terraform Version
<!--- Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed. -->
1.4.2

### Affected Resource(s)
<!--- Please list the resources as a list, for example: -->
- `davinci_flow`


<!--- If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this. -->

### Terraform Configuration Files
N/A

### Panic Output
```log
╷
│ Error: Plugin did not respond
│ 
│   with davinci_flow.portal_login,
│   on dev.tf line 19973, in resource "davinci_flow" "portal_login":
│ 19973: resource "davinci_flow" "portal_login"{
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).PlanResourceChange call. The plugin logs may contain
│ more details.
╵

Stack trace from the terraform-provider-davinci_v0.1.13 plugin:
panic: interface conversion: interface {} is nil, not map[string]interface {}
goroutine 883 [running]:
github.com/pingidentity/terraform-provider-davinci/internal/service/davinci.expandSubFlowProps(0xc0001349c0?)
	github.com/pingidentity/terraform-provider-davinci/internal/service/davinci/resource_flow.go:774 +0x439
github.com/pingidentity/terraform-provider-davinci/internal/service/davinci.mapSubFlows(0xcc5720?, {0xc000d80000, 0x24b95})
	github.com/pingidentity/terraform-provider-davinci/internal/service/davinci/resource_flow.go:736 +0x272
github.com/pingidentity/terraform-provider-davinci/internal/service/davinci.computeFlowDrift({0x4e601f?, 0xcfdd88?}, {0xc000c5c000, 0x19922}, {0xc000d80000, 0x24b95}, 0xc00005fa00?)
	github.com/pingidentity/terraform-provider-davinci/internal/service/davinci/resource_flow.go:526 +0xcf
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diff(0x6?, {0xe27320, 0xc000448d80}, {0xcfdd88, 0x9}, 0xc00043c8c0, 0xc00005f980, {0xe2a0f0?, 0xc00005fa00}, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/schema.go:1143 +0x39b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Diff(0xc00042bbc0, {0xe27320, 0xc000448d80}, 0xc000144c30, 0xc000137110, 0x0, {0xce5860, 0xc00028b680}, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/schema.go:[678](-/jobs/5591562852#L678) +0x353
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).SimpleDiff(0xe27a58?, {0xe27320?, 0xc000448d80?}, 0xc000144c30, 0xc0c640?, {0xce5860?, 0xc00028b[680](-/jobs/5591562852#L680)?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:962 +0xe5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).PlanResourceChange(0xc0003fbaa0, {0xe27320?, 0xc000448b10?}, 0xc0006a3220)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:780 +0xa09
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).PlanResourceChange(0xc0004f5d60, {0xe27320?, 0xc00042b080?}, 0xc00041e1c0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:824 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_PlanResourceChange_Handler({0xcc6fe0?, 0xc0004f5d60}, {0xe27320, 0xc00042b080}, 0xc00041e150, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:449 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004c05a0, {0xe2afb8, 0xc0005829c0}, 0xc000420240, 0xc0004482d0, 0x1341120, 0x0)
	google.golang.org/grpc@v1.57.0/server.go:1360 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc0004c05a0, {0xe2afb8, 0xc0005829c0}, 0xc000420240, 0x0)
	google.golang.org/grpc@v1.57.0/server.go:1737 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.57.0/server.go:982 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.57.0/server.go:980 +0x18c
Error: The terraform-provider-davinci_v0.1.13 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior
<!--- What should have happened? -->
Should apply terraform plan

### Actual Behavior
<!--- What actually happened? -->
It crashed

### Steps to Reproduce
<!---Please list the steps required to reproduce the issue, for example: -->
1. create a flow json and name the flow with something > limit chars
2. `terraform apply`

### Important Factoids
<!--- Are there anything you'd like to share about the general setup of your PingOne account?  Please do not include sensitive information or account data -->
The flow refused to import in to DaVinci console until I reduced the length of the flow name. Not sure if this is the only reason for the crash, however it seems that a flow/subflow with a long/descriptive name causes a panic

--- a/internal/provider/provider.go
+++ b/internal/provider/provider.go
@@ -156,13 +156,10 @@ func retryableClient(cInput *client.ClientInput) (*client.APIClient, error) {
 			// These cases come from the davinci-client-go library and may be subject to change
 		case strings.Contains(err.Error(), "Error getting admin callback, got: status: 502, body:"):
 			tflog.Info(ctx, "Found retryable error while initializing client. Retrying...")
-			fmt.Printf("Sign in retryable Error: %s\n", err.Error())
 		case strings.Contains(err.Error(), "Error getting SSO callback, got err: status: 502, body:"):
 			tflog.Info(ctx, "Found retryable error while initializing client. Retrying...")
-			fmt.Printf("Sign in retryable Error: %s\n", err.Error())
 		case strings.Contains(err.Error(), "Auth Token not found, unsuccessful login, got: Found. Redirecting to https://console.pingone.com/davinci/index.html#/sso/callback/?error=AuthenticationFailed&error_description=unknownError2"):
 			tflog.Info(ctx, "Found retryable error while initializing client. Retrying...")
-			fmt.Printf("Sign in retryable Error: %s\n", err.Error())
 		default:
 			return nil, err
 		}

--- a/internal/service/davinci/data_source_application_test.go
+++ b/internal/service/davinci/data_source_application_test.go
@@ -3,7 +3,6 @@ package davinci_test
 import (
 	"fmt"
 	"testing"
-	"time"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/pingidentity/terraform-provider-davinci/internal/acctest"
@@ -11,11 +10,6 @@ import (
 
 // TestAccDatasourceApplication_Slim - Depends on testAccResourceApplication_Slim_Hcl
 func TestAccDatasourceApplication_SlimByAppId(t *testing.T) {
-	// tflog.Info(ctx, "Rate limit hit, retrying...")
-	timeElapsed := 4 * time.Second
-	fmt.Println(timeElapsed)
-	fmt.Println(timeElapsed)
-
 	resourceBase := "davinci_application"
 	resourceName := acctest.ResourceNameGen()
 	dataSourceFullName := fmt.Sprintf("data.%s.%s", resourceBase, resourceName)

--- a/internal/service/davinci/resource_application_flow_policy.go
+++ b/internal/service/davinci/resource_application_flow_policy.go
@@ -168,7 +168,6 @@ func resourceApplicationFlowPolicyRead(ctx context.Context, d *schema.ResourceDa
 	}
 
 	flatResp, err := flattenAppPolicy(resp, policyId)
-	// fmt.Printf("flatResp: %+v\n", flatResp)
 	if err != nil {
 		if strings.Contains(err.Error(), "Unable to find policy with id") {
 			d.SetId("")

--- a/internal/service/davinci/resource_connection_test.go
+++ b/internal/service/davinci/resource_connection_test.go
@@ -175,7 +175,6 @@ func TestAccResourceConnection_HeavyRead(t *testing.T) {
 	// resourceFullName := fmt.Sprintf("%s.%s", resourceBase, resourceName)
 
 	// hcl := testAccResourceConnection_HeavyRead_Hcl(resourceName, "heavy")
-	// fmt.Printf(`HCL: \n %s \n`, hcl)
 	testAccResourceConnection_HeavyRead_Hcl(resourceName, "heavy")
 
 	// resource.ParallelTest(t, resource.TestCase{

--- a/internal/service/davinci/resource_flow.go
+++ b/internal/service/davinci/resource_flow.go
@@ -795,17 +795,28 @@ func mapSubFlows(d *schema.ResourceData, flowJson string) (*string, error) {
 
 func expandSubFlowProps(subflowProps map[string]interface{}) (*dv.SubFlowProperties, error) {
 
-	sfp := subflowProps["subFlowId"].(map[string]interface{})
-	sfpVal := sfp["value"].(map[string]interface{})
+	sfp, ok := subflowProps["subFlowId"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("Flow Validation Error: subFlowId key not found in subflow properties")
+	}
+	sfpVal, ok := sfp["value"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("Flow Validation Error: subFlowId value not found in subflow properties")
+	}
 	sfId := dv.SubFlowID{
 		Value: dv.SubFlowValue{
 			Value: sfpVal["value"].(string),
 			Label: sfpVal["label"].(string),
 		},
 	}
-	subflowVersionId := subflowProps["subFlowVersionId"].(map[string]interface{})
+	subflowVersionId, ok := subflowProps["subFlowVersionId"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("Flow Validation Error: subFlowVersionId not found in subflow properties")
+	}
 	var sfvidString string
-
+	if subflowVersionId["value"] == nil {
+		return nil, fmt.Errorf("Flow Validation Error: subFlowVersionId.value not found in subflow properties")
+	}
 	switch subflowVersionId["value"].(type) {
 	case int:
 		sfvidString = strconv.Itoa(subflowVersionId["value"].(int))
@@ -814,14 +825,14 @@ func expandSubFlowProps(subflowProps map[string]interface{}) (*dv.SubFlowPropert
 	case string:
 		sfvidString = subflowVersionId["value"].(string)
 	default:
-		return nil, fmt.Errorf("Error: subflow versionId is not a string or int")
+		return nil, fmt.Errorf("Flow Validation Error: subflow versionId is not a string or int")
 	}
 
 	sfv := dv.SubFlowVersionID{
 		Value: sfvidString,
 	}
 	if sfId.Value.Value == "" || sfv.Value == "" {
-		return nil, fmt.Errorf("Error: subflow value or versionId is empty")
+		return nil, fmt.Errorf("Flow Validation Error: subflow value or versionId is empty")
 	}
 	subflow := dv.SubFlowProperties{
 		SubFlowID:        sfId,
@@ -886,7 +897,15 @@ func validateFlowDeps(d *schema.ResourceData, diags *diag.Diagnostics) {
 			// validate subflows
 			if v.Data.ConnectorID == "flowConnector" && (v.Data.CapabilityName == "startSubFlow" || v.Data.CapabilityName == "startUiSubFlow") {
 				foundSubflow := false
-				sfProp, _ := expandSubFlowProps(v.Data.Properties)
+				sfProp, err := expandSubFlowProps(v.Data.Properties)
+				if err != nil {
+					*diags = append(*diags, diag.Diagnostic{
+						Severity: diag.Error,
+						Summary:  "Error Validating flow_json",
+						Detail:   err.Error(),
+					})
+					return
+				}
 				if subflows, ok := d.GetOk("subflow_link"); ok {
 					sfList := subflows.(*schema.Set).List()
 					for _, sfMap := range sfList {

